### PR TITLE
fix: Persist uuid from SA call in `stack_analyses_request` table

### PR DESCRIPTION
**Pull request by rafiu007**
This pr makes chnages in the api server so it can store the uuid from sa calls into the newly created user_id column in stack_analyses_request table.

### CODE REVIEW

File: bayesian/utility/v2/stack_analyses.py [hunk 1]
```diff
@@ -55,13 +57,15 @@ def post_request(self):
         # Generate unique request id using UUID, also record timestamp in readable form
         self._new_request_id = str(uuid.uuid4().hex)
         date_str = str(datetime.datetime.now())
+        # Fetch uuid from header
+        header_data = HeaderData(uuid=request.headers.get('uuid', None))
```
Comment (prashbnair) [inline bayesian/utility/v2/stack_analyses.py:61]: why do you need to use HeaderData. you can store the uuid in a variable.

File: bayesian/utility/v2/stack_analyses.py [hunk 2]
```diff
@@ -55,13 +57,15 @@ def post_request(self):
...
 
         # Make backbone request
         deps = self._make_backbone_request()
 
         # Finally save results in RDS and upon success return request id.
         rdbAnalyses = RdbAnalyses(self._new_request_id, date_str,
-                                  self._manifest_file_info, deps)
+                                  self._manifest_file_info, deps, header_data.uuid)
```
Comment (prashbnair) [inline bayesian/utility/v2/stack_analyses.py:68]: where is the code to save the field.

File: bayesian/utility/db_gateway.py [hunk 3]
```diff
@@ -217,12 +217,13 @@ class RdbAnalyses:
     Provides interfaces to save and read request post data for stack analyses v2.
     """
 
-    def __init__(self, request_id, submit_time=None, manifest=None, deps=None):
+    def __init__(self, request_id, submit_time=None, manifest=None, deps=None, user_id=None):
```
Comment (prashbnair) [inline bayesian/utility/db_gateway.py:220]:
```suggestion
    def __init__(self, request_id):
```

Change the constructor to only take in parameters that are used by all functions.  The other parameters are specific to the save_post_request method and can be passed as parameters to that.

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: bayesian/utility/db_gateway.py (hunk 3 above)
Comment (rafiu007) [reply bayesian/utility/db_gateway.py:220]: Okay .Makes sense.

Comment (rafiu007) [reply bayesian/utility/db_gateway.py]: Resolved.

**Review by prashbnair — APPROVED** [review]

**Pull request merged**